WordPress/src/main/res/layout/fragment_login_site_check_error.xml

```diff
@@ -1,25 +1,69 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    android:orientation="vertical"
-    android:layout_height="match_parent"
-    android:layout_width="match_parent">
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_gravity="center"
+    android:fillViewport="true">
 
-    <include
-        layout="@layout/toolbar_login"/>
-
-    <ScrollView
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
Suggestion (💡): There exist a lot of duplication here, that is, in relation to the `fragment_login_no_sites_empty_view.xml` file. Maybe, we should extract these duplicate views into a separate layout file and reuse that new layout here and there.

Actually, thinking a bit more about it, it seems to me that `LoginNoSitesFragment` and `LoginSiteCheckErrorFragment` share a lot of functionality as well, thus duplicating a lot of code. The main diff as far as I can see is the `ARG_SITE_ADDRESS`. As such, maybe those two fragments and functionality needs to be merged into one and then their corresponding layout files to be merged as well. I feel that this will reduce a lot of duplication and make any future maintainability more manageable and less error prone to, for example, updating one view while forgetting to update the other one as well.

Wdyt?

Reply:
@ParaskP7 - I see your POV and I thought about it earlier, but made the choice to keep them separate. What it comes down to is this: Is using `if/case` statements to separate responsibility better than separate classes? One can argue this point from both sides :). Same goes for hiding/showing elements in a layout. You can argue that not having to hide/show makes it less error prone because you don't have to remember which case should hide or show elements. I decided to keep them individual fragment/layouts so that each had a single responsibility. However, I am open to the changing to meet the expectations of the majority. :)

Reply:
@zwarm , totally with you on that, my only fear is the amount of duplication and maintainability overhead, or even problems with that. Other than that, I agree with your point, that complexity just (kind of) shifts with one solution vs. the other.